diff --git forkSrcPrefix/Makefile.libretro forkDstPrefix/Makefile.libretro
index ffaba21e7a88f90786d818ab550b68acd05738bf..6ff526a5a0a77b4ebaf0ffbd38cc78a9e93b901d 100644
--- forkSrcPrefix/Makefile.libretro
+++ forkDstPrefix/Makefile.libretro
@@ -381,6 +381,20 @@ else ifeq ($(platform), gcw0)
    CFLAGS += -fomit-frame-pointer -ffast-math -march=mips32 -mtune=mips32r2 -mhard-float
    CXXFLAGS += $(CFLAGS)
 
+# Miyoo A30
+else ifeq ($(platform), my282)
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
+	SHARED := -shared -Wl,--version-script=$(version_script)
+	LDFLAGS += -fPIC -flto
+	PLATFORM_DEFINES := -DCC_RESAMPLER -DCC_RESAMPLER_NO_HIGHPASS
+	CFLAGS += -marm -mtune=cortex-a7
+	CFLAGS += -mfpu=neon-vfpv4 -mfloat-abi=hard
+	CFLAGS += -fomit-frame-pointer -ffast-math -fPIC -flto
+	CXXFLAGS += $(CFLAGS)
+
 # RETROFW
 else ifeq ($(platform), retrofw)
    TARGET := $(TARGET_NAME)_libretro.so
